src/UI/Seller/src/app/promotions/components/promotion-edit/promotion-edit.component.ts

```diff
@@ -488,6 +489,18 @@ export class PromotionEditComponent implements OnInit, OnChanges {
     )
   }
 
+  isBtnDisabled(): boolean {
+    if (!this.areChanges && this.isCreatingNew) {
+      if (this.resourceForm?.status === 'INVALID' || this.dataIsSaving) {
+        return true
+      }
+    } else if (!this.areChanges && !this.isCreatingNew) {
+      return true
```

Comment:
Shouldn't this be flip-flopped with the above? If there are no changes and its creating a new promo then there wouldn't be any data and we should assume the button should be disabled. And conversely, if there are no changes but its editing an existing promo then we'll need to rely on form validation to know if the button should be disabled.

Can you also rename to a more descriptive name like `isSaveBtnDisabled`